File: matlab/res_alloc_stream_data.m
% resource allocation for stream processing.

rand('state',1);
randn('state',1);

J = 10; % number of job types
P = 20; % number of process types
n = 5; % number of system resources

R = sprandn(P,J,0.6);
R = abs(R) > 0;  % job process matrix

x_tot = 10*(1+rand(n,1)); % total resources available
A = 3+7*rand(n,P);
x_min = 0.2*rand(n,P);  % minimum resources to run each process
w = 1+rand(J,1);
t_tar = 2+8*rand(J,1);  % target job traffic
l_max = 1+2*rand(J,1); % maximum allowed job latency

File: python/res_alloc_stream_data.py
# resource allocation for stream processing.
import numpy as np
import scipy.io as sio

def get_value(mat, name):
    value = mat[name]
    return value[0][0] if np.shape(value) == (1, 1) else value

mat = sio.loadmat("res_alloc_stream_data.mat")

J = get_value(mat, "J")  # number of job types
P = get_value(mat, "P")  # number of process types
n = get_value(mat, "n")  # number of system resources

R = get_value(mat, "R").astype(float)  # job process matrix
x_tot = get_value(mat, "x_tot")  # total resources available
A = get_value(mat, "A")
x_min = get_value(mat, "x_min")  # minimum resources to run each process
w = get_value(mat, "w")
t_tar = get_value(mat, "t_tar")  # target job traffic
l_max = get_value(mat, "l_max")  # maximum allowed job latency
